rpc_differ/rpc_differ.py

```diff
@@ -134,29 +141,37 @@ def create_parser():
     return parser
 
 
-def get_osa_commit(repo, ref):
+def get_osa_commit(repo, ref, rpc_product=None):
     """Get the OSA sha referenced by an RPCO Repo."""
     repo.head.reference = repo.commit(ref)
     repo.head.reset(index=True, working_tree=True)
-    try:
-        osa_commit = repo.submodules['openstack-ansible'].hexsha
-    except IndexError:
+
+    functions_path = os.path.join(repo.working_tree_dir,
+                                  'scripts/functions.sh')
+    release_path = os.path.join(repo.working_tree_dir,
+                                'playbooks/vars/rpc-release.yml')
+
+    if os.path.exists(release_path):
+        with open(release_path) as f:
+            rpc_release_data = yaml.safe_load(f.read())
+
+        release_data = rpc_release_data['rpc_product_releases'][rpc_product]
+        return release_data['osa_release']
+    elif os.path.exists(functions_path):
         # This branch doesn't use a submodule for OSA
```

Comment:
Since deriving the value from the functions file is the most broken, I think it should be last, with submodules being the first fallback after the release file.

Reply:
done.